target/linux/ath79/dts/qca9563_glinet_gl-ar750s.dtsi

```diff
@@ -163,14 +176,34 @@
 	};
 };
 
-&eth0 {
+&pcie {
 	status = "okay";
+};
 
-	mtd-mac-address = <&art 0x0>;
-	phy-handle = <&phy0>;
+&uart {
+	status = "okay";
+};
+
+&usb0 {
+	status = "okay";
+
+	vbus-supply = <&usb_vbus>;
```

Comment:
This line can be dropped. qca9563 usb uses generic ehci drivers which has no support for vbus-supply. As you can see usb_vbus is an always-on regulator, which means it activates on its own.

Reply:
The GL-AR750S [needs GPIO7 pulled to supply VBUS to the USB connector](https://docs.gl-inet.com/en/3/hardware/ar750s/).

My earlier versions of this PR used `gpio-hog` to pull GPIO7 (going back to at least May).

I agree that the USB driver has no support for managing a VBUS regulator and that, as far as I can tell `regulator-fixed` has no user-space method of controlling its on/off state. As such, use of `regulator-fixed` offers no benefits over `gpio-hog`. Use of `regulator-fixed` requires adding to _all_ ath79-nand kernels:
```
+CONFIG_REGULATOR=y
+CONFIG_REGULATOR_FIXED_VOLTAGE=y
```
with no end-user benefit that I can perceive for the increased kernel size.

The `regulator-fixed` was added in commit 0f6b944c923 as a result of https://github.com/openwrt/openwrt/pull/2065

I raised concerns about use of `regulator-fixed` over `gpio-hog` at that time. However that #2065 was merged using `regulator-fixed`. 

In deference to the decision to accept #2065, its author, and its committer, I did not change it to `gpio-hog` in my commit sequence.

I still believe that use of `gpio-hog` should be preferred in this case as it results in a smaller kernel, is consistent with [upstream Linux guidance](https://www.kernel.org/doc/Documentation/devicetree/bindings/gpio/gpio.txt), and, as far as I can tell, does not impact utility in any way, as compared to `regulator-fixed'

Should you agree, I would be happy to remove the otherwise-unused `CONFIG_REGULATOR` kernel config parameters<sup>1</sup> for the ath79-nand kernel, replace with use of a gpio-hog, and test proper operation.


<sup>1</sup> Note that the usages of "regulator" are not otherwise found in the ath79-nand devices
```
openwrt$ fgrep -lr regulator target/linux/ath79/dts/
target/linux/ath79/dts/ar9331_pisen_wmm003n.dts
target/linux/ath79/dts/ar9331_tplink_tl-wr710n-v1.dts
target/linux/ath79/dts/ar7242_avm_fritz300e.dts
target/linux/ath79/dts/ar9331_tplink_tl-mr3040-v2.dts
target/linux/ath79/dts/ar9331_tplink_tl-mr3020-v1.dts
target/linux/ath79/dts/qca9563_glinet_gl-ar750s.dts
target/linux/ath79/dts/ar9331_tplink_tl-wr703n_tl-mr10u.dtsi
target/linux/ath79/dts/qca9531_tplink_tl-wr810n-v1.dts
target/linux/ath79/dts/ar9344_qihoo_c301.dts
```
```
openwrt$ fgrep 'define Device' target/linux/ath79/image/nand.mk 
define Device/aerohive_hiveap-121
define Device/glinet_gl-ar300m-nand
define Device/netgear_ath79_nand
define Device/netgear_wndr4300
define Device/zyxel_nbg6716
```

Reply:
Will switch to `gpio-hog` and remove regulator-related kernel config from NAND kernel in next push.

Saves ~14 kB in compressed kernel size with no apparent reduction in functionality.

With my "usual" build:
```
openwrt-ath79-nand$ ./scripts/diffconfig.sh | fgrep kmod
CONFIG_PACKAGE_kmod-crypto-crc32c=y
CONFIG_PACKAGE_kmod-crypto-hash=y
CONFIG_PACKAGE_kmod-fs-ext4=y
CONFIG_PACKAGE_kmod-lib-crc16=y
```

1,952,020 kernel size with `regulator-fixed`
1,937,164 kernel size with `gpio-hog`

14,856 savings

Detection of USB stick in USB socket confirms that supply is present.
```
[   11.409623] usb-storage 2-1.1:1.0: USB Mass Storage device detected
[   11.433692] scsi host1: usb-storage 2-1.1:1.0
[   11.978528] scsi 0:0:0:0: Direct-Access     SanDisk  Extreme          0001 PQ: 0 ANSI: 6
[   11.989449] sd 0:0:0:0: [sda] 31277232 512-byte logical blocks: (16.0 GB/14.9 GiB)
[   11.998260] sd 0:0:0:0: [sda] Write Protect is off
[   12.003223] sd 0:0:0:0: [sda] Mode Sense: 53 00 00 08
[   12.008076] sd 0:0:0:0: [sda] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[   12.022408]  sda: sda1 sda2
```